platforms/ide/ide-plugins/src/main/java/org/gradle/plugins/ide/eclipse/EclipseWtpPlugin.java

```diff
@@ -284,11 +286,11 @@ public void execute(JavaPlugin javaPlugin) {
                 ((IConventionAware) eclipseModel.getWtp().getFacet()).getConventionMapping().map("facets", new Callable<List<Facet>>() {
                     @Override
                     public List<Facet> call() throws Exception {
-                        return Lists.newArrayList(
-                            new Facet(Facet.FacetType.fixed, "jst.java", null),
-                            new Facet(Facet.FacetType.installed, "jst.utility", "1.0"),
-                            new Facet(Facet.FacetType.installed, "jst.java", toJavaFacetVersion(project.getExtensions().getByType(JavaPluginExtension.class).getSourceCompatibility()))
-                        );
+                        List<Facet> result = new ArrayList<>(3);
```

Comment:
💭 Is something modifying the return value from an anonymous type that promises only some `List`, nothing definitely modifiable?  Can we use immutable list here and avoid the modification?

Reply:
`EclipseWtpFacet` is part of an extension on the Project. It is a user-modifyable model, and should be mutable